### ISSUE TITLE
Generating classes instead of enums

### ISSUE THREAD

**Issue by man-csl** [joelittlejohn/jsonschema2pojo#1129]
I am trying to generate classes for OCPP2 specification using maven. Everything works except that the generator creates classes instead of enums for enum types and the code does not compile.

See below for an example:
```
public class APNAuthenticationEnum {

    @SerializedName("CHAP")
    CHAP("CHAP"),
    @SerializedName("NONE")
    NONE("NONE"),
    @SerializedName("PAP")
    PAP("PAP"),
    @SerializedName("AUTO")
    AUTO("AUTO");
....
}
```

The json specification :

```
"APNAuthenticationEnumType": {
      "description": "APN. APN_ Authentication. APN_ Authentication_ Code\r\nurn:x-oca:ocpp:uid:1:568828\r\nAuthentication method.\r\n",
      "javaType": "APNAuthenticationEnum",
      "type": "string",
      "additionalProperties": false,
      "enum": [
        "CHAP",
        "NONE",
        "PAP",
        "AUTO"
      ]
    },

```

I already had a look in to some of the existing issues mentioning this problem, but in most cases the suggestion was to change the  javaType as a work around. In my case the specification is external so dont want change it. Appreciate if some one could please let me know the options I have.

**Comment by lars-andren** [comment]
Hear hear!
Experiencing the same problem when doing the exact same thing, but with Gradle.

**Comment by lars-andren** [comment]
Esteemed hacker Hannes Nyberg contributed the following;

"It works better if you first add a package name to all javaType in the schema, here's an example script;

export PACKAGE=ocpp20
export STRING_TO_REPLACE='\"javaType\": \"'
export REPLACEMENT=$STRING_TO_REPLACE$PACKAGE'.'echo "Preparing json files"
cp -r json json-workingFILES_DIR=$(pwd)
echo "Scanning files in $FILES_DIR/json-working/"
for filename in "$FILES_DIR"/json-working/*.json
do
	#echo "Preparing file: $filename"
	sed -i "s/$STRING_TO_REPLACE/$REPLACEMENT/g" $filename
done

"

**Comment by eirnym** [comment]
My bet that `additionalProperties` field prevents generating enums and it's quite weird for `string` type. 

So it's a double bug: the `jsonschema2pojo` generates classes because of this field, and specification itself incorrectly has `additionalProperties` for type `string`

**Comment by manushawijekoon** [comment]
removing additionalProperties did not fix this.

**Comment by unkish** [comment]
Issue is due to `javaType` not being specified with package name (see #1002)

**Comment by joelittlejohn** [comment]
The problem here was a bug in CodeModel. When enums go into the default package (no package) they become classes. This is fixed in v1.1.0. See #1103.

**Comment by ostfale** [comment]
Using latest version (v1.1.0) creates another problem for OCPP 2.0.1 Json schema files: 

`
[ERROR] Could not create enum.
org.jsonschema2pojo.exception.ClassAlreadyExistsException
    at org.jsonschema2pojo.rules.EnumRule.createEnum (EnumRule.java:363)
    at org.jsonschema2pojo.rules.EnumRule.apply (EnumRule.java:117)
...
`
